tests/dragonfly/cluster_test.py

```diff
@@ -907,272 +972,196 @@ async def test_random_keys():
 
-    await push_config(
-        config.replace("LAST_SLOT_CUTOFF", "2999").replace("NEXT_SLOT_CUTOFF", "3000"),
-        c_nodes_admin,
-    )
+    nodes[0].migrations = []
+    nodes[0].slots = [(0, 2999)]
+    nodes[1].slots = [(3000, 16383)]
+    logging.debug("remove finished migrations")
+    await push_config(json.dumps(generate_config(nodes)), [node.admin_client for node in nodes])
 
-    assert await c_nodes[0].get("KEY0") == "value"
-    assert await c_nodes[1].get("KEY1") == "value"
-    assert await c_nodes[1].get("KEY2") == "value"
-    assert await c_nodes[1].get("KEY3") == "value"
-    assert await c_nodes[0].get("KEY4") == "value"
-    assert await c_nodes[1].get("KEY5") == "value"
-    assert await c_nodes[1].get("KEY6") == "value"
-    assert await c_nodes[1].get("KEY7") == "value"
-    assert await c_nodes[0].get("KEY8") == "value"
-    assert await c_nodes[1].get("KEY9") == "value"
-    assert await c_nodes[1].get("KEY10") == "value"
-    assert await c_nodes[1].get("KEY11") == "value"
-    assert await c_nodes[1].get("KEY12") == "value"
-    assert await c_nodes[1].get("KEY13") == "value"
-    assert await c_nodes[1].get("KEY14") == "value"
-    assert await c_nodes[1].get("KEY15") == "value"
-    assert await c_nodes[1].get("KEY16") == "value"
-    assert await c_nodes[1].get("KEY17") == "value"
-    assert await c_nodes[1].get("KEY18") == "value"
-    assert await c_nodes[1].get("KEY19") == "value"
-    assert await c_nodes[1].get("KEY20") == "value"
-    assert await c_nodes[1].get("KEY21") == "value"
-    assert await c_nodes[1].execute_command("DBSIZE") == 19
+    assert await nodes[0].client.get("KEY0") == "value"
+    assert await nodes[1].client.get("KEY1") == "value"
+    assert await nodes[1].client.get("KEY2") == "value"
+    assert await nodes[1].client.get("KEY3") == "value"
+    assert await nodes[0].client.get("KEY4") == "value"
+    assert await nodes[1].client.get("KEY5") == "value"
+    assert await nodes[1].client.get("KEY6") == "value"
+    assert await nodes[1].client.get("KEY7") == "value"
+    assert await nodes[0].client.get("KEY8") == "value"
+    assert await nodes[1].client.get("KEY9") == "value"
+    assert await nodes[1].client.get("KEY10") == "value"
+    assert await nodes[1].client.get("KEY11") == "value"
+    assert await nodes[1].client.get("KEY12") == "value"
+    assert await nodes[1].client.get("KEY13") == "value"
+    assert await nodes[1].client.get("KEY14") == "value"
+    assert await nodes[1].client.get("KEY15") == "value"
+    assert await nodes[1].client.get("KEY16") == "value"
+    assert await nodes[1].client.get("KEY17") == "value"
+    assert await nodes[1].client.get("KEY18") == "value"
+    assert await nodes[1].client.get("KEY19") == "value"
+    assert await nodes[1].client.get("KEY20") == "value"
+    assert await nodes[1].client.get("KEY21") == "value"
+    assert await nodes[1].client.execute_command("DBSIZE") == 19
```

Comment:
for k in (f'KEY{i}' for i in range(20)):
  assert nodes[crc(key) // 8000] == value

screams to me 😆

Reply:
BRING the party parots on github! https://cultofthepartyparrot.com/